core/src/main/java/org/elasticsearch/search/internal/ShardSearchLocalRequest.java

```diff
@@ -167,6 +176,11 @@ protected void innerReadFrom(StreamInput in) throws IOException {
         source = in.readOptionalWriteable(SearchSourceBuilder::new);
         types = in.readStringArray();
         aliasFilter = new AliasFilter(in);
+        if (in.getVersion().onOrAfter(Version.V_5_2_0_UNRELEASED)) {
+            indexBoost = in.readFloat();
+        } else {
+            indexBoost = 1.0f;
```

Comment:
nice! fix looks good! did you have some tests out of `qa/backwards-5.0` fail without the fix? Or did you just test manually? The backwards tests run the rest tests against a mixed version cluster, so I *think* your newly added tests should fail in that setup. Can you verify? If so we are good to go!